src/tests/hostpath/hostpath_data_integrity/hostpath_data_integrity_test.go

```diff
@@ -0,0 +1,104 @@
+package hostpath_data_integrity
+
+import (
+	"testing"
+	"time"
+
+	"github.com/openebs/openebs-e2e/common"
+	"github.com/openebs/openebs-e2e/common/e2e_ginkgo"
+	"github.com/openebs/openebs-e2e/common/hostpath"
+
+	"github.com/openebs/openebs-e2e/common/k8stest"
+	coreV1 "k8s.io/api/core/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestHostpathDataIntegrityTest(t *testing.T) {
+	// Initialise test and set class and file names for reports
+	e2e_ginkgo.InitTesting(t, "hostpath_data_integrity", "hostpath_data_integrity")
+}
+func dataIntegrityTestTest(decor string, fstype common.FileSystemType) {
+	app := k8stest.FioApplication{
+		Decor:                   decor,
+		VolSizeMb:               1024,
+		OpenEbsEngine:           common.Hostpath,
+		VolType:                 common.VolFileSystem,
+		FsType:                  fstype,
+		Loops:                   1,
+		VolWaitForFirstConsumer: true,
+		AddFioArgs:              []string{"--size=500m", "--status-interval=30"},
+	}
+
+	hostptahConfig, err := hostpath.GetHostPathAnnotationConfig()
+	Expect(err).To(BeNil(), "failed to get hostpath annotation config")
+	Expect(hostptahConfig).ToNot(BeNil(), "hostpath annotation config should not be nil")
+
+	// setup sc parameters
+	app.HostPath = k8stest.HostPathOptions{
+		Annotations: hostptahConfig,
+	}
+	if app.FsType == common.BtrfsFsType {
+		app.FsPercent = 60
+	}
+	// create sc, pvc, fio pod
+	logf.Log.Info("create sc, pvc, fio pod")
+	err = app.DeployApplication()
+	Expect(err).To(BeNil(), "failed to deploy app")
+
+	// Check fio pod status
+	phase, podLogSysnopsis, err := k8stest.CheckFioPodCompleted(app.GetPodName(), common.NSDefault)
+	Expect(err).To(BeNil(), "CheckPodComplete got error %s", err)
+	Expect(phase).ShouldNot(Equal(coreV1.PodFailed), "fio pod phase is %s, %s", phase, podLogSysnopsis)
+
+	logf.Log.Info("Waiting for fio to complete", "timeoutSecs", 240)
+	_, err = app.WaitFioComplete(240, 30)
```

Comment:
Wouldn't app.WaitComplete be more appropriate here? then the pod can be deleted immediately

Reply:
updated with app.WaitComplete https://github.com/openebs/openebs-e2e/compare/c1fb225e3d4d43369ace798570c6a8ca064a5418..462e99a8e99047d8de274daa5701e92473cb97a0